zerver/tests/test_events.py

```diff
@@ -247,6 +247,13 @@ def verify_action(self,
         )
         action()
         events = client.event_queue.contents()
+        contents: List[Dict[str, Any]] = []
+        for event in events:
+            if event['type'] == 'message' and 'internal_data' in event.keys():
+                del event['internal_data']
+            contents.append(event)
+        events = contents
```

Comment:
I think we should have this call the same `prune_internal_data` function I suggest extracting below, rather than copying the code.